spec/unit/outputs/elasticsearch_spec.rb

```diff
@@ -73,6 +93,27 @@
       end
     end
 
+    context "on a reachable ES instance" do
+      let(:events) { [ ::LogStash::Event.new("foo" => "bar1"), ::LogStash::Event.new("foo" => "bar2") ] }
+
+      let(:shutdown_value) { true }
+
+      let(:logger) { double("logger") }
+      before(:each) do
+        allow(subject).to receive(:logger).and_return(logger)
+        allow(logger).to receive(:info)
+
+        allow(subject).to receive(:pipeline_shutdown_requested?) do
+          shutdown_value
+        end
```

Comment:
Until it's necessary I prefer to be explicit to facilitate reading the code:

```suggestion
      let(:logger) { double("logger") }

      before(:each) do
        allow(subject).to receive(:logger).and_return(logger)
        allow(logger).to receive(:info)

        allow(subject).to receive(:pipeline_shutdown_requested?).and_return(true)
```